org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -16,63 +16,99 @@ namespace MixedReality.Toolkit.Input
     [AddComponentMenu("MRTK/Input/Near Interaction Mode Detector")]
     public class NearInteractionModeDetector : ProximityDetector
     {
+        /// <summary>
+        /// The set of near interactables that belongs to near interaction
+        /// </summary>
         [SerializeField]
-        [Tooltip("The set of near interactors that belongs to near interaction")]
-        private List<XRBaseInteractor> nearInteractors;
+        [Tooltip("The set of near interactables that belongs to near interaction")]
+        private List<XRBaseInteractable> nearInteractables;
 
         /// <summary>
-        /// Used to keep track of the previously detected colliders so that we can know which
-        /// colliders stopped being detected and update their buttons front plate and rounded
-        /// rect if they are labeled as dynamic (based on proximity).
+        /// Used to keep track of the previously detected interactrables so that we can know which
+        /// interactable stopped being detected and update their buttons front plate RawImage.
         /// </summary>
-        List<Collider> previouslyDetectedColliders = new List<Collider>();
+        private HashSet<IXRProximityInteractable> previouslyDetectedColliderInteractableDuples = new();
 
         /// <inheritdoc />
         public override bool IsModeDetected()
         {
             bool result = base.IsModeDetected() || IsNearInteractorSelecting();
             if (result)
             {
-                for (int i = 0; i < previouslyDetectedColliders.Count; i++)
+                UpdateProximityExited();
+                UpdateProximityEntered();
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Calls OnProximityExited for all interactables that were previously ProximityHover detected but are no longer detected.
+        /// </summary>
+        private void UpdateProximityExited()
+        {
+            List<IXRProximityInteractable> currentlyDetectedInteractable = GetCurrentlyDetectedColliderInteractablesDuples();
+            List<IXRProximityInteractable> noLongerDetectedInteractables = new();
```

Comment:
we should not be allocating a new list every frame.   Allocations have performance impact when done every frame.

If you want to do this "copy to list" pattern.   Make the list a class member, and clear it every frame instead of allocating.

Why did the reverse iteration not work?

Reply:
I missed that.  Moved the datastructures to class-level variables in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/659/commits/7ee0fe33c7946426a3a643e4c266e4defd1fc3e2
Re: Why did the reverse iteration not work? ... I think it was working, however, since there was feedback from another reviewer that there were missing colliders from processing I decided to reduce the number moving parts in the refactoring and go the safe route.  Would you like me to retry with a reverse iteration?
In the meantime, resolving this conversation (please feel free to open a new conversation if you would like me to tackle the reverse iteration approach) since the issue of allocating a new list every frame is addressed in https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/659/commits/7ee0fe33c7946426a3a643e4c266e4defd1fc3e2